widgetssdk/src/test/java/com/glia/widgets/core/permissions/PermissionManagerTest.kt

```diff
@@ -0,0 +1,472 @@
+import com.glia.widgets.permissions.PermissionsRequestRepository
+import com.glia.widgets.permissions.PermissionsRequestResult
+import org.junit.Assert.assertEquals
+import org.junit.Before
+import org.junit.Test
+import org.mockito.kotlin.any
+import org.mockito.kotlin.argumentCaptor
+import org.mockito.kotlin.eq
+import org.mockito.kotlin.mock
+import org.mockito.kotlin.never
+import org.mockito.kotlin.verify
+import org.mockito.kotlin.whenever
+
+class PermissionManagerTest {
+    private lateinit var context: Context
+    private lateinit var checkSelfPermission: CheckSelfPermission
+    private lateinit var permissionsRequestRepository: PermissionsRequestRepository
+
+    private lateinit var permissionManager: PermissionManager
+
+    @Before
+    fun setUp() {
+        context = mock()
+        checkSelfPermission = mock()
+        permissionsRequestRepository = mock()
+
+        permissionManager = PermissionManager(
+            context,
+            checkSelfPermission,
+            permissionsRequestRepository,
+            Build.VERSION_CODES.R
+        )
+    }
+
+    // .getPermissionsForEngagementMediaType()
+
+    @Test
+    fun `getPermissionsForEngagementMediaType returns empty lists of permissions when mediaType is TEXT`() {
+        val result = permissionManager.getPermissionsForEngagementMediaType(Engagement.MediaType.TEXT, false)
+
+        assertEquals(Permissions(emptyList(), emptyList()), result)
+    }
+
+    @Test
+    fun `getPermissionsForEngagementMediaType returns empty lists of permissions when mediaType is AUDIO`() {
+        val result = permissionManager.getPermissionsForEngagementMediaType(Engagement.MediaType.AUDIO, false)
+
+        assertEquals(Permissions(
+            listOf(Manifest.permission.RECORD_AUDIO),
+            emptyList()
+        ), result)
+    }
+
+    @Test
+    fun `getPermissionsForEngagementMediaType returns empty lists of permissions when mediaType is VIDEO`() {
+        val result = permissionManager.getPermissionsForEngagementMediaType(Engagement.MediaType.VIDEO, false)
+
+        assertEquals(Permissions(
+            listOf(Manifest.permission.CAMERA, Manifest.permission.RECORD_AUDIO),
+            emptyList()
+        ), result)
+    }
+
+    @Test
+    fun `getPermissionsForEngagementMediaType returns empty lists of permissions when mediaType is AUDIO, isCallVisualizer`() {
+        val result = permissionManager.getPermissionsForEngagementMediaType(Engagement.MediaType.AUDIO, true)
+
+        assertEquals(Permissions(emptyList(), emptyList()), result)
+    }
+
+    @Test
+    fun `getPermissionsForEngagementMediaType returns empty lists of permissions when mediaType is VIDEO, isCallVisualizer`() {
+        val result = permissionManager.getPermissionsForEngagementMediaType(Engagement.MediaType.VIDEO, true)
+
+        assertEquals(Permissions(
+            listOf(Manifest.permission.CAMERA),
+            emptyList()
+        ), result)
+    }
+
+    //
```

Comment:
I suppose you wanted to add some comment here, didn't you?

Reply:
Fixed